packages/react/src/ActionList/Group.tsx

```diff
@@ -85,8 +85,10 @@ export const Group: React.FC<React.PropsWithChildren<ActionListGroupProps>> = ({
     >
       <GroupContext.Provider value={{selectionVariant, groupHeadingId}}>
         {title && !slots.groupHeading ? (
-          <GroupHeading title={title} variant={variant} auxiliaryText={auxiliaryText} />
+          // Escape hatch: supports old API <ActionList.Group title="group title"> in a non breaking way
+          <GroupHeading variant={variant} auxiliaryText={auxiliaryText} unsafeTitle={title} />
```

Comment:
To make sure that the invariant check doesn't go through on the old API so that it doesn't break any usecases in downstream repos, I kept the `ActionList.GroupHeading` sub component's API flexible.  Especially, to distinguish where the group heading is coming from i.e. old api vs new api. 

If it comes from old api (i.e. if it has title prop), I render the heading like

```
<GroupHeading variant={variant} auxiliaryText={auxiliaryText} title={title} />
```

if it uses the new api (if children exists), 

```
<GroupHeading variant={variant} auxiliaryText={auxiliaryText} >{title}</GroupHeading>
```

Reply:
Optional name idea, instead of `unsafeTitle`: `_internalBackwardCompatibleTitle`